services/api-server/src/simcore_service_api_server/models/schemas/solvers.py

```diff
@@ -104,3 +105,36 @@ def resource_name(self) -> str:
     @classmethod
     def compose_resource_name(cls, solver_key, solver_version) -> str:
         return compose_resource_name("solvers", solver_key, "releases", solver_version)
+
+
+PortKindStr = Literal["input", "output"]
+
+
+class SolverPort(BaseModel):
+    key: str = Field(
+        ...,
+        description="port identifier name",
+        regex=PUBLIC_VARIABLE_NAME_RE,
+        title="Key name",
+    )
+    kind: PortKindStr
+    content_schema: Optional[dict[str, Any]] = Field(
+        None,
+        description="jsonschema for the port's value. SEE https://json-schema.org",
+    )
+
+    class Config:
+        extra = Extra.ignore
+        schema_extra = {
+            "example": {
+                "key": "input_2",
```

Comment:
so it's key as well. was just confused because of the PR description 👍

Reply:
true. I forgot to change the example in the PR